qcelemental/models/molecule.py

```diff
@@ -105,26 +110,28 @@ class Molecule(ProtoModel):
         ...,
         # LNN Note: "Ordered" might not be the least ambiguous term
         # (e.g. "ordered" as in linked ordering or "ordered" as in alphabetical?)
-        description="An ordered 1-D array-like object of atomic elemental symbols of shape (N). Index of this list "
+        description="An ordered 1-D array-like object of atomic elemental symbols of shape (NAtom). Index of this list "
                     "sets atomic order for all other per-atom setting like ``real`` and the first dimension of "
-                    "``geometry``. Ghost/virtual atoms "
-                    "must have an entry in this array-like and are indicated by the matching index in ``real``"
-                    "field."
+                    "``geometry``. Ghost/Virtual atoms must have an entry in this array-like and are indicated by the "
+                    "matching index in ``real`` field."
     )
     geometry: Array[float] = Schema(
         ...,
-        description="An ordered 2-D array-like object of shape (N,3) for atomic coordinates. Index of the "
-                    "first dimension matches the index of all other per-atom settings like ``symbols`` and ``real``."
+        description="An ordered 2-D array-like object of shape (NAtom,3) for XYZ atomic coordinates [a0]. "
+                    "Atom ordering is fixed; that is, a consumer who shuffles atoms must not reattach the input "
+                    "(pre-shuffling) molecule schema instance to any output (post-shuffling) per-atom results "
+                    "(e.g., gradient). Index of the first dimension matches the index of all other per-atom settings "
+                    "like ``symbols`` and ``real``."
```

Comment:
I think we may be conflating documentation targets. There's internal storage (ndarray `(nat, 3)`). There's what can be input to the object constructor (array-like that somehow total 3*nat). There's serialization format (`(3*nat, )` if msgpack unavailable, if I understand the new stuff correctly). Should we be clear about target audience context?

Reply:
Good question. I think the main user who will interact with these docs are those which will actually either be constructing these objects themselves (`Molecule(geometries=...)`) or those who are working with the constructed objects (`mol.geometries[1,:]`) , so I think targeting those users are the ones we should worry about rather than the sterilization crowd. 

I do agree there is a bit ambiguity because someone can feed in a 1-D array-like of length `3*NAtom` and it will be converted to the `(NAtom, 3)` shape. I can clean that part up if you think that will help

Reply:
Maybe `geometry` only gets a shout-out to the serialization crowd b/c that was (strangely) a main bullet result of the MolSSI schema workshop.

But, yes, I think outright stating the internal storage in first sentence/paragraph, then in second paragraph expanding on allowed init values is a good separation. It'll be especially handy for chgmult, where internal always has a value, while upon init you can give all, nothing, or in between.